lib/berkeley_library/docker/railtie.rb

```diff
@@ -2,10 +2,28 @@
 
 module BerkeleyLibrary
   module Docker
-    class Railtie < Rails::Railtie
-      NAME = 'berkeley_library-docker.load_secrets'.freeze
+    class Railtie < ::Rails::Railtie
+      include Logging
 
-      initializer(NAME, after: :initialize_logger) { Secret.load_secrets! }
+      def secrets
+        @secrets ||= {}
+      end
+
+      def load_secrets!
+        @secrets = Secret.load_secrets!
+      end
+
+      private
+
+      config.before_configuration { load_secrets! }
```

Comment:
I wrote a unit test which exercises this, but have also confirmed that it works with our internal Framework release (currently pinned to 0.2.0.pre.rc2). This actually gets called before any inline `config.foo = 'bar'` statements in application.rb.